src/rest/parser.hpp

```diff
@@ -80,8 +80,25 @@ class Parser
     void Process(const char *aBuf, size_t aLength);
 
 private:
+    class State {
+    public:
```

Comment:
nit:
```suggestion
    struct State {
```

Let's juts use `struct` for data classes